src/monster.cpp

```diff
@@ -195,9 +195,23 @@ void monster::poly( const mtype_id& id )
     faction = type->default_faction;
     upgrades = type->upgrades;
 }
+void monster::acquire_xp(int amount){
+  Creature::acquire_xp(amount);
+  if(has_flag(MF_EXPERIENCE_UPGRADE)){
+    if(type->half_life<=get_xp()){
+      static const mtype_id mon_null( "mon_null" );
+      if(type->upgrade_into!=mon_null){
+	poly(type->upgrade_into);
+      }else{
+	const std::vector<mtype_id> monsters = MonsterGroupManager::GetMonstersFromGroup(type->upgrade_group);
+	poly(random_entry(monsters));//this seems like the only option with pokemon like eevee.
+      }
+    }	
+  }
+}
 
 bool monster::can_upgrade() {
-    return upgrades && (ACTIVE_WORLD_OPTIONS["MONSTER_UPGRADE_FACTOR"] > 0.0);
+  return upgrades && (ACTIVE_WORLD_OPTIONS["MONSTER_UPGRADE_FACTOR"] > 0.0)&&!has_flag(MF_EXPERIENCE_UPGRADE);
```

Comment:
Yes. It made sense for the pokemon mod, though as I'm removing that from this pull request, I might remove that too, if this is something that is preferable.